### PR TITLE
Implementing accountant fee

### PR DESCRIPTION
**Pull request by chompomonim**

### CODE REVIEW

File: contracts/AccountantImplementation.sol [hunk 1]
```diff
@@ -57,23 +65,26 @@ contract AccountantImplementation is FundsRecovery {
...
     */
 
     // Because of proxy pattern this function is used insted of constructor.
     // Have to be called right after proxy deployment.
-    function initialize(address _token, address _operator) public {
+    function initialize(address _token, address _operator, uint16 _fee) public {
         require(!isInitialized(), "have to be not initialized");
         require(_operator != address(0), "operator have to be set");
         require(_token != address(0), "token can't be deployd into zero address");
+        require(_fee <= 5000, "fee can't be bigger that 50%");
```
Comment (vkuznecovas) [inline contracts/AccountantImplementation.sol]:
```suggestion
        require(_fee <= 5000, "fee can't be bigger than 50%");
```

File: contracts/AccountantImplementation.sol [hunk 2]
```diff
@@ -134,8 +145,11 @@ contract AccountantImplementation is FundsRecovery {
         // Increase already paid amount
         _channel.settled = _channel.settled.add(_unpaidAmount);
 
+        // Canculate accountant fee
```
Comment (vkuznecovas) [inline contracts/AccountantImplementation.sol]:
```suggestion
        // Calculate accountant fee
```

File: contracts/AccountantImplementation.sol [hunk 3]
```diff
@@ -288,6 +302,29 @@ contract AccountantImplementation is FundsRecovery {
         emit ChannelBeneficiaryChanged(_channelId, _newBeneficiary);
     }
 
+    function setAccountantFee(uint16 _newFee, bytes memory _signature) public {
+        require(_newFee <= 5000, "fee can't be bigger that 50%");
+        require(block.number >= lastFee.validFrom, "can't update not active fee");
```
Comment (vkuznecovas) [inline contracts/AccountantImplementation.sol]:
```suggestion
        require(block.number >= lastFee.validFrom, "can't update inactive fee");
```

File: contracts/AccountantImplementation.sol (hunk 1 above)
Comment (zolia) [inline contracts/AccountantImplementation.sol]: What is 5000 ? 5k tokens? why 5k? How come its 50%?

File: contracts/AccountantImplementation.sol [hunk 4]
```diff
@@ -288,6 +302,29 @@ contract AccountantImplementation is FundsRecovery {
...
+
+        if (msg.sender != operator) {
+            address _signer = keccak256(abi.encodePacked(UPDATE_FEE_PREFIX, address(this), _newFee)).recover(_signature);
+            require(_signer == operator, "have to be signed by accountant operator");
+        }
+
+        // new fee will start be valid after delay block will pass
```
Comment (zolia) [inline contracts/AccountantImplementation.sol]: style.

File: docs/smart-contracts.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ Main functions of Registry:
 `registerAccountant` function (in payload 0xf4c1a1f6) is registering new accountant and requires such attributes:
  * `_accountantOperator`: address of account which will operate accountant smart contract and sign all off-chain messages on his behalf.
  * `_stakeAmount`: amount of tokens (in wei) to be staked by accountant as security deposit.
+ * `_accountantFee`: fee in subprocents (2.5% = 250) which will be charged during promise settlement (in provider channel).
```
Comment (zolia) [inline docs/smart-contracts.md]:
```suggestion
 * `_accountantFee`: fee in fractional percentage (2.5% = 250) which will be charged during promise settlement (in provider channel).
```

File: contracts/AccountantImplementation.sol (hunk 1 above)
Comment (chompomonim) [reply contracts/AccountantImplementation.sol]: 1% = 1.00 = 100 sub procent parts
50% = 50.00 = 5000 subprocent parts

Comment (chompomonim) [reply contracts/AccountantImplementation.sol]: We're working only with integers and I want to have possibility to charge 0.05% fee.

**Pull request merged**